packages/medusa/src/services/custom-shipping-option.ts

```diff
@@ -0,0 +1,106 @@
+import { MedusaError } from "medusa-core-utils"
+import { EntityManager } from "typeorm"
+import { TransactionBaseService } from "../interfaces"
+import { CustomShippingOption } from "../models"
+import { CustomShippingOptionRepository } from "../repositories/custom-shipping-option"
+import { FindConfig, Selector } from "../types/common"
+import { CreateCustomShippingOptionInput } from "../types/shipping-options"
+import { buildQuery } from "../utils"
+
+class CustomShippingOptionService extends TransactionBaseService<CustomShippingOptionService> {
+  protected manager_: EntityManager
+  protected transactionManager_: EntityManager | undefined
+  protected customShippingOptionRepository_: typeof CustomShippingOptionRepository
+
+  constructor({ manager, customShippingOptionRepository }) {
+    // eslint-disable-next-line prefer-rest-params
```

Comment:
yeah, I am a bit cautious about adding it to the global lint config since it would apply globally. Since we only use this specifically in the constructor of services, I dont think it's the biggest deal to have this line in the constructor of each service, but it would be nice to disable it only in constructors if possible. 

Otherwise we could consider creating a custom rule to enforce this only outside of the constructors, and specifically to just ignore only the `super(arguments[0])` call.

Reply:
when there is not 10th injected components, I suggest to just pass them normally `super({ manager, ... })`